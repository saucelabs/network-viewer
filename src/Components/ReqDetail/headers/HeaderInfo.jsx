import React, { useState } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';

import Styles from './../Headers.styles.scss';
import HeaderTitle from './HeaderTitle';

const context = classNames.bind(Styles);

const HeaderInfo = ({
  eventKey,
  data,
  component,
  isEncodeEnabled,
  isParseEnabled,
}) => {
  const [isVisible, updateVisibleStates] = useState(true);
  const [isPayloadTransformed, updateTransform] = useState(true);

  const handlePayloadTransform = () => updateTransform(!isPayloadTransformed);
  const ChildComponent = () => component({
    data,
    isPayloadTransformed,
    onChangeEncode: handlePayloadTransform,
  });

  return (
    <div className={context('header-info', { active: isVisible })}>
      <HeaderTitle
        eventKey={eventKey}
        isEncodeEnabled={isEncodeEnabled}
        isParseEnabled={isParseEnabled}
        isPayloadTransformed={isPayloadTransformed}
        onClick={() => updateVisibleStates(!isVisible)}
        onPayloadTransform={handlePayloadTransform}
      />
      {isVisible && <ChildComponent /> }
    </div>
  );
};

HeaderInfo.propTypes = {
  component: PropTypes.func.isRequired,
  data: PropTypes.object,
  eventKey: PropTypes.string.isRequired,
  isEncodeEnabled: PropTypes.bool,
  isParseEnabled: PropTypes.bool,
};

HeaderInfo.defaultProps = {
  data: null,
  isEncodeEnabled: false,
  isParseEnabled: false,
};

export default HeaderInfo;
