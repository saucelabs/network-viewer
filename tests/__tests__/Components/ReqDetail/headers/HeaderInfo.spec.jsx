import React from 'react';
import { shallow } from 'enzyme';

import HeaderInfo from './../../../../../src/Components/ReqDetail/headers/HeaderInfo';

describe('HeaderInfo', () => {
  const props = {
    eventKey: 'general',
    data: {},
    component: () => {},
  };

  it('renders without crashing', () => {
    const element = shallow(<HeaderInfo {...props} />);
    expect(element).toMatchSnapshot();
  });
});
