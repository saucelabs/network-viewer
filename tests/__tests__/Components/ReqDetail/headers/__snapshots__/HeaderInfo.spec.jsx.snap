// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderInfo renders without crashing 1`] = `
<div
  className="header-info"
>
  <HeaderTitle
    eventKey="general"
    isEncodeEnabled={false}
    isParseEnabled={false}
    isPayloadTransformed={true}
    onClick={[Function]}
    onPayloadTransform={[Function]}
  />
</div>
`;
